test/utils/SlotDerivation.t.sol

```diff
@@ -3,18 +3,20 @@
 
 pragma solidity ^0.8.20;
 
-import {Test} from "forge-std/Test.sol";
+// solhint-disable func-name-mixedcase
 
+import {Test} from "forge-std/Test.sol";
+import {SymTest} from "halmos-cheatcodes/SymTest.sol";
 import {SlotDerivation} from "@openzeppelin/contracts/utils/SlotDerivation.sol";
 
-contract SlotDerivationTest is Test {
+contract SlotDerivationTest is Test, SymTest {
     using SlotDerivation for bytes32;
 
     bytes[] private _array;
 
-    function testDeriveArray(uint256 length, uint256 offset) public {
-        length = bound(length, 1, type(uint256).max);
-        offset = bound(offset, 0, length - 1);
+    function testSymbolicDeriveArray(uint256 length, uint256 offset) public {
+        vm.assume(length > 0);
+        vm.assume(offset < length);
```

Comment:
Using `vm.assume` drops fuzzing branches, and thus reduces the size of the fuzzing "coverage". `bound` is the recommanded option when. Any good reason for that change?

Reply:
I see that Halmos recommend using vm.assume. 

It may be better to have 2 version, one with vm.assume for halmos, and one with bound for fuzzing

Reply:
Yeah good observation, I splitted the test in two so that the fuzzing version keeps `bound`